### DISCUSSION TITLE
Dictionary of agents by type

### DISCUSSION THREAD

**Discussion by kulkarnipreetam** [mesa/mesa#2179] [Q&A] [answered]
Hi,

I have multiple agent types in my model. I wanted to know if there is a way to access a dictionary by agent types in which the agent's unique id is the key and the agent instance (memory address) is the value so that I can use the agent's unique id to access the attributes of the agent?

This is similar to what a "who" number with "ask" statement would do in Netlogo. The reason I need this is because if I have agents of type A and B, I need to process agent A's data (agents filtered/sorted based on an attribute) in agent  B's functions/methods.

Thank you :)

**Comment by kulkarnipreetam** [comment, marked as answer]
I found this... 
self.schedule.agents_by_type["Agent class name"]

**Comment by EwoutH** [comment]
Sorry for not getting back to you, it seems we all missed it.

In Mesa there's a function `get_agents_of_type()` that gives you an `AgentSet` with all the agents of a certain type.

https://github.com/projectmesa/mesa/blob/3cf1b76ca3a157c8a064b60130755a86e73718be/mesa/model.py#L107-L109

Let me know if that helps!
